### PR TITLE
Add .eslintignore file

### PR DESCRIPTION
**Pull request by pdehaan**
Use .eslintignore file instead of `--ignore-pattern` CLI args.

### CODE REVIEW

**Comment by openjck** [comment]
Thanks! Can we add _docs/_ to the file? We'll also need to remove the `--ignore-pattern` that was just added by a conflicting commit.

File: .eslintrc [hunk 1]
```diff
@@ -1,3 +1,5 @@
+extends: eslint:recommended
```
Comment (openjck) [inline .eslintrc:1]: Ah, neat. I didn't know about that.

Comment (groovecoder) [inline .eslintrc:1]: Are any of the lines in here made obsolete by extending recommended?

**Comment by pdehaan** [comment]
Rebased and squashed.

File: .eslintrc (hunk 1 above)
Comment (pdehaan) [inline .eslintrc:1]: Excellent question...

| Rule | Value | [`eslint:recommended`](https://github.com/eslint/eslint/blob/master/conf/eslint.json) | Notes |
| --- | :-: | :-: | --- |
| `curly` | 0 | "off" / 0 | Same. |
| `eqeqeq` | 2 | "off" / 0 | &nbsp; |
| `guard-for-in` | 2 | "off" / 0 | &nbsp; |
| `indent` | [2, 4, {SwitchCase: 1}] | "off" / 0 | &nbsp; |
| `no-bitwise` | 2 | "off" / 0 | &nbsp; |
| `no-caller` | 2 | "off" / 0 | &nbsp; |
| `no-empty` | 2 | "error" / 2 | Same. |
| `no-extend-native` | 2 | "off" / 0 | &nbsp; |
| `no-irregular-whitespace` | 2 | "error" / 2 | Same. |
| `no-undef` | 2 | "error" / 2 | Same. |
| `no-unused-vars` | 2 | "error" / 2 | Same. |
| `quotes` | [2, single] | "off" / 0 | &nbsp; |
| `strict` | [2, function] | "off" / 0 | &nbsp; |

Looks like we have 5 rules that are currently the default in eslint:recommended. We can remove those from our `rules` in .eslintrc, or leave them if we are concerned that they ever may change in future versions of the recommended rules.

Comment (openjck) [inline .eslintrc:1]: I would be slightly in favor of keeping our options. `eslint:recommended` is a great baseline, but I wouldn't want our explicit choices to come to depend on the choices of the ESLint team in the future.

Comment (groovecoder) [inline .eslintrc:1]: Hmm ... I started to agree with you, but I also like following conventions over configurations. 

I.e., if the ESLint team (and community) changes their rules in a way that breaks our front-style checks, it's a trigger to us to decide - either follow the ESLint community convention or keep our existing convention.

Comment (openjck) [inline .eslintrc:1]: That's true, assuming we notice of course. Either way works for me. I doesn't sound likely that `eslint:recommended` will change anyway.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "scripts": {
-    "test": "eslint --ignore-pattern 'docs/**/*.js' --ignore-pattern 'staticfiles/**/*.js' --ignore-pattern 'env/**/*.js' . && stylint **/static/styles/"
+    "test": "eslint . && stylint **/static/styles/*.styl"
```
Comment (openjck) [inline package.json]: Looks like the `*.styl` part got accidentally restored during the rebase. As per #165 we'll need to keep that out.

Comment (pdehaan) [inline package.json]: Fixed. Good catch.

File: .eslintrc (hunk 1 above)
Comment (pdehaan) [inline .eslintrc:1]: So, do we want me to remove the 5 duplicated rules, or are we cool to merge as-is?

Comment (openjck) [inline .eslintrc:1]: I'm cool to merge as-is. Testing now...

**Comment by openjck** [comment]
Looks good, thanks @pdehaan!

**Pull request merged**